ggplot2switch

mbstadler opened the pull request:
Closes #82 

Thanks @machlabd for getting this started!
I am opening a PR so we and @csoneson can use it to coordinate our efforts.
Of course at the moment it is not ready to merge yet - I am marking it as "Draft"
and adding you two as reviewers so we won't inadvertently merge it before we
all think it's ready.

machlabd committed:
update my email in Description

machlabd committed:
started modifying plotStabilityPaths() to depend on ggplot2. ToDo: add checks, review test functions, update Description file

machlabd committed:
add ggplot2 dependency to description file and remove "invisible(TRUE)" return for plotStabilityPaths()

machlabd committed:
adjust (for now minimal) test for plotStabilityPaths()()

machlabd committed:
arrange package dependencies alphabetically in Description file (for easier track-keeping)

machlabd committed:
update .Rd files. ToDo: additional checks for plotStabilityPaths()

machlabd committed:
add missing comma in Description file

machlabd committed:
add missing tidyr dependency

machlabd committed:
add checks for plotStabilityPaths()

mbstadler committed:
Merge branch 'devel' into ggplot2switch

mbstadler committed:
update docs after merge devel

mbstadler commented:
Here is an overview of all plots in `monaLisa`, so we can tick off the ones that have been switched to `ggplot2`:

- [x] `plotBinDiagnostics`
    - [x] `aspect == "length"`
    - [x] `aspect == "GCfrac"`
    - [x] `aspect == "dinucfreq"` (not switching, based on `ComplexHeatmap`)
- [x] `plotBinDensity`
- [x] `plotBinHist`
- [x] `plotBinScatter`
- [x] `plotMotifHeatmaps` (not switching, based on `ComplexHeatmap`)
- [x] `plotSelectionProb`
- [x] `plotStabilityPaths`

Other todo's before merging:
- [x] remove `vioplot` from imports/depends
- [x] update static plot pngs used in `README.md` and man pages:
    - [x] `man/figures/monaLisa_binning_small.png`
    - [x] `man/figures/monaLisa_heatmaps_small.png` (`plotMotifHeatmaps` is unchanged, no update needed)
- [x] update `NEWS.md`
- [x] bump version
- [x] check how/if the new changes affect plots in the vignettes and adjust if necessary

mbstadler committed:
update .assertVector and .assertScalar (cli is also imported by ggplot2)

mbstadler committed:
return luminance in addition to cols in attributes of value of getColsByBin

mbstadler committed:
update tests for .assertScalar and .assertVector

mbstadler committed:
switch to testthat edition 3

mbstadler committed:
remove unnecessary LazyData from DESCRIPTION

mbstadler committed:
replace vioplot by ggplot2::geom_violin in plotBinDiagnostics

mbstadler committed:
use rlang::.data to suppress warnings about missing bindings for global variables

mbstadler committed:
use cli_abort instead of stop

csoneson committed:
Expose glmnet arguments for stability selection

mbstadler committed:
use .message (based on cli) instead of message to report on progress

mbstadler committed:
use ggplot2::geom_histogram instead of base graphics for plotBinHist

mbstadler committed:
use ggplot2::geom_area instead of base graphics for plotBinDensity

mbstadler committed:
fix typo

mbstadler committed:
improve test coverage

mbstadler committed:
use ggplot2::geom_point instead of base graphics for plotBinScatter

mbstadler committed:
ensure sequences have names before writing to file

mbstadler committed:
improve test coverage

mbstadler committed:
use rlang::.data to avoid warnings about unknown global variables

mbstadler committed:
use theme_classic for consistency with plotStabilityPaths

csoneson commented on NAMESPACE:
```diff
@@ -27,6 +27,7 @@ export(randLassoStabSel)
 export(sampleRandomRegions)
 export(seqLogoGrob)
 export(setZeroBin)
+import(ggplot2)
```
At the moment we import both all of ggplot2 and the individual functions below - I don't think we need both.

mbstadler replied:
I agree. It seems that the `roxygen2` documentation (https://roxygen2.r-lib.org/articles/namespace.html?q=import#imports) recommends to use `@importFrom` over `@import`. I'll change it accordingly.

csoneson commented on R/plotBinDiagnostics.R:
```diff
@@ -14,62 +14,85 @@
-#'   the output of \code{vioplot()}, which generates the plot. For
-#'   aspect=\code{"dinucfreq"}, returns (invisibly) the ComplexHeatmap
-#'   object.
+#' @return For aspect=\code{"length"} or \code{"GCfrac"}, returns \code{ggplot2}
```
-> returns *a* \code{ggplot2} object

mbstadler replied:
Fixed (I changed it to `returns a \code{\link[ggplot2]{ggplot}} object`, as there is no `ggplot2` class or object.

csoneson commented on R/plotBinDiagnostics.R:
```diff
@@ -14,62 +14,85 @@
-                               horizontal = TRUE, cex = 0.5, cex.axis = 0.75,
-                               cex.names = 0.75, par(mar = c(4, 6, 2, 2) + 0.1))
-        return(invisible(vp))
+        p <- ggplot(data = data.frame(seqlen = width(seqs),
```
Two thoughts on this chunk: 
* the code for the 'length' and 'GC' plots are essentially the same, the only difference being the x-axis (data and label). What about combining them (defining the data and label depending on the `aspect`, to avoid code repetition? 
* I'm not totally sure about the luminance-dependent line colour. I think it is nice for the quantiles, but I feel that it could possibly be misleading for the violin outline (left plot below uses black in all cases, right plot the luminance-dependent color). 
<img width="1385" alt="Screenshot 2025-02-26 at 18 49 05" src="https://github.com/user-attachments/assets/e58d3729-eefd-49f9-b3cb-9636afe4d79e" />

mbstadler replied:
I fully agree. I have combined the 'length' and 'GC' plots as much as possible, and removed the luminance-based line coloring to avoid the issue above.

csoneson commented on R/plotting.R:
```diff
@@ -64,54 +71,78 @@ getColsByBin <- function(b,
+#'     If not \code{"none"}, draw a legend with binning information. The value
+#'     is used to control the legend position and will be passed to
+#'     \code{theme(legend.position = legendPosition)}.
+#' @param legend Depreciated (ignored). Please use \code{legendPosition} to
```
I believe this (and line 86) should say "Deprecated" rather than "Depreciated"

mbstadler replied:
Fixed (I think I have been contaminated by budget planning thinking...).

csoneson commented on R/plotting.R:
```diff
@@ -64,54 +71,78 @@ getColsByBin <- function(b,
+    .assertScalar(x = ylab, type = "character")
+    .assertScalar(x = main, type = "character")
+    if (!is.null(legend)) {
+        cli_warn(c("{.arg legend} is depreciated and ignored.",
```
Also here (and line 124), depreciated -> deprecated

mbstadler replied:
Fixed

csoneson commented on R/plotting.R:
```diff
@@ -64,54 +71,78 @@ getColsByBin <- function(b,
+    }
+    if (!is.null(legend.cex)) {
+        cli_warn(c("{.arg legend.cex} is depreciated and ignored.",
+                   "i" = "You can use {.fn theme} to control legend and other graphical paramters"))
```
paramters -> parameters

mbstadler replied:
Fixed

csoneson commented on R/plotting.R:
```diff
@@ -120,173 +151,251 @@ plotBinHist <- function(x, b, breaks = 10 * nlevels(b),
+#'     If not \code{"none"}, draw a legend with binning information. The value
+#'     is used to control the legend position and will be passed to
+#'     \code{theme(legend.position = legendPosition)}.
+#' @param legend Depreciated (ignored). Please use \code{legendPosition} to
```
Depreciated -> Deprecated

mbstadler replied:
Fixed

csoneson commented on R/plotting.R:
```diff
@@ -120,173 +151,251 @@ plotBinHist <- function(x, b, breaks = 10 * nlevels(b),
+    .assertScalar(x = ylab, type = "character")
+    .assertScalar(x = main, type = "character")
+    if (!is.null(legend)) {
+        cli_warn(c("{.arg legend} is depreciated and ignored.",
```
depreciated -> deprecated

mbstadler replied:
Fixed

csoneson commented on R/plotting.R:
```diff
@@ -120,173 +151,251 @@ plotBinHist <- function(x, b, breaks = 10 * nlevels(b),
+    }
+    if (!is.null(legend.cex)) {
+        cli_warn(c("{.arg legend.cex} is depreciated and ignored.",
+                   "i" = "You can use {.fn theme} to control legend and other graphical paramters"))
```
paramters -> parameters

mbstadler replied:
Fixed

csoneson commented on R/plotting.R:
```diff
@@ -120,173 +151,251 @@ plotBinHist <- function(x, b, breaks = 10 * nlevels(b),
+#'     If not \code{"none"}, draw a legend with binning information. The value
+#'     is used to control the legend position and will be passed to
+#'     \code{theme(legend.position = legendPosition)}.
+#' @param legend Depreciated (ignored). Please use \code{legendPosition} to
```
Depreciated -> Deprecated

mbstadler replied:
Fixed

csoneson commented on R/plotting.R:
```diff
@@ -120,173 +151,251 @@ plotBinHist <- function(x, b, breaks = 10 * nlevels(b),
-    invisible(TRUE)
+    if (!is.null(legend.cex)) {
+        cli_warn(c("{.arg legend.cex} is depreciated and ignored.",
+                   "i" = "You can use {.fn theme} to control legend and other graphical paramters"))
```
paramters -> parameters

mbstadler replied:
Fixed

csoneson commented on R/plotting.R:
```diff
@@ -120,173 +151,251 @@ plotBinHist <- function(x, b, breaks = 10 * nlevels(b),
+            "cannot use custom colors ({.arg cols}) with bin-based legend"))
+    }
+    if (!is.null(legend)) {
+        cli_warn(c("{.arg legend} is depreciated and ignored.",
```
depreciated -> deprecated

mbstadler replied:
Fixed

csoneson commented on R/plotting.R:
```diff
@@ -510,72 +620,98 @@ plotMotifHeatmaps <- function(x,
-#' @return \code{TRUE} (invisibly).
-#' 
-#' @examples 
+#' @param col vector of colors specifying what to color selected and
```
This sentence is a bit hard to parse. Perhaps can also clarify that it should be a named vector with names 'sel' and 'notSel'

machlabd replied:
good point! I was wondering if there was a more elegant way to do this as well.

csoneson replied:
I think the other option might be to just have two arguments - e.g. `selColor` and `notSelColor` or something like that, and let the user specify these directly.

machlabd replied:
yes that nicer!

csoneson commented on R/plotting.R:
```diff
@@ -510,72 +620,98 @@ plotMotifHeatmaps <- function(x,
 #'
 #' @importFrom SummarizedExperiment assay rowData colData
-#' @importFrom graphics matplot
+#' @import ggplot2
```
Could we import specific functions from ggplot2 also here for consistency?

mbstadler replied:
Done

csoneson commented on R/plotting.R:
```diff
@@ -510,72 +620,98 @@ plotMotifHeatmaps <- function(x,
+        cli_abort("{.arg se} must be a {.cls SummarizedExperiment}")
     }
+    .assertScalar(x = selProbMin, type = "numeric", rngIncl = c(0, 1))
+    .assertVector(x = col, type = "character", len = 2L)
```
Check here that the names are the expected ones

csoneson commented on R/plotting.R:
```diff
@@ -510,72 +620,98 @@ plotMotifHeatmaps <- function(x,
+    .assertVector(x = ylim, type = "numeric", rngIncl = c(0, 1))
+    .assertVector(x = colnames(se), type = "character", len = ncol(se))
+    .assertVector(x = rownames(se), type = "character", len = nrow(se))
+    .assertVector(x = colnames(colData(se)), type = "character")
```
Check already here that there are some columns starting with `regStep` (expected below)?

machlabd replied:
Yeah I also wondered if I should add that. `regStep` should in principle be there, since it happens when calling `randLassoStabSel`, and the output of that should be fed into the plotting function. But can't hurt to have that extra check!

csoneson reviewed (commented):
Thanks @mbstadler and @machlabd! I have added some comments in the code, but overall I think this looks good!

csoneson committed:
Merge branch 'ggplot2switch' into expose-glmnet-args

csoneson committed:
Render documentation

csoneson committed:
Add tests

mbstadler committed:
fix typos

mbstadler committed:
remove luminance-based line coloring (makes extreme values invisible and violins hard to compare)

mbstadler committed:
combine redundant code fro aspects "length" and "GCfrac"

mbstadler committed:
fix more typos

mbstadler committed:
use @importFrom instead of @import

csoneson reviewed (approved):
Changes look great, thanks :)

csoneson committed:
Merge branch 'ggplot2switch' into expose-glmnet-args

csoneson committed:
Ignore minAbsX if its outide the data range

csoneson committed:
Merge pull request #87 from fmicompbio/expose-glmnet-args

Expose glmnet arguments (e.g., family) for stability selection

machlabd commented on NAMESPACE:
```diff
@@ -109,7 +109,7 @@ importFrom(ggplot2,labs)
 importFrom(ggplot2,theme)
-importFrom(ggplot2,theme_bw)
+importFrom(ggplot2,theme_classic)
 importFrom(glmnet,glmnet)
```
I am also happy to swicth to `theme_bw` if you prefer. I was undecided between the two :)

mbstadler replied:
Either is fine for me :-) I like that they are both quite calm

machlabd committed:
working draft of plotSelectionProb()

machlabd committed:
resolved merge conflicts

machlabd committed:
separated `col` into two params for `plotStabilityPaths()` and added check for "regStep" in colnames.

machlabd committed:
add `selProbCutoffColor` argument to `plotStabilityPaths()`

machlabd committed:
create .assertColor and add to `plotStabilityPaths()`

machlabd committed:
added separate color argument for the cutoff line in `plotSelectionProb()`

machlabd committed:
stick to `importFrom` instead of `import` in `plotSelectionProb()`

machlabd committed:
remove `importFrom graphics` in `plotSelectionProb()`

machlabd committed:
fix tests for `plotSelectionProb()`

machlabd committed:
removed 'importFrom graphics` at top of plotting.R file

machlabd committed:
fixed figure size in `plotSelProbs`chunk of stability selection vignette

machlabd commented:
I think I am done making changes to the stability selection plots for now. One more idea I am toying with, is (optionally) adding labels for the selected predictors in the `plotStabilityPaths` function. However, that would introduce a dependency on `ggrepel`..
Here's a depiction of what I was playing with: 
```
gg <- plotStabilityPaths(se)
gg + ggrepel::geom_text_repel(data = subset(df, selected & regStep == max(regStep)), aes(label = predictor), nudge_x = 3, na.rm = TRUE, size = 3, family = "Lato", direction = "y", color = "cadetblue", hjust = 0, segment.linetype = "dotted", segment.size = 0.7, segment.curvature = -0.1, segment.angle = 20, box.padding = 0.4, segment.alpha = 0.5, vjust = 0) + coord_cartesian(clip = 'off')
```
![plotStabilityPathsWithLabels](https://github.com/user-attachments/assets/4e3d1158-fba0-4ec0-bad6-01bbd691c9fa)

I am tempted to leave this to avoid adding more package dependencies. On the other hand if this could be useful then we could add it.
(This was inspired by this: https://r-graph-gallery.com/web-line-chart-with-labels-at-end-of-line.html)

mbstadler commented:
I think the `ggrepel` version of `plotStabilityPaths` is amazing (maybe limiting the axis and selProbMin horizontal line to the range of data?).

One way to avoid making `monaLisa` too dependency heavy would be to add `ggrepel` to `Suggests` and then add an argument to `plotStabilityPaths(..., labelPaths = FALSE)`. Only if `labelPaths = TRUE`, you would then `.assertPackagesAvailable("ggrepel")` (see https://github.com/fmicompbio/swissknife/blob/master/R/utils.R) to make sure it is installed. As long as you go with the default `labelPaths = FALSE`, the function can run without `ggrepel` being installed.

mbstadler commented:
@machlabd By the way - feel free to tick off above in the todo list all the boxes that you have finished :-)

machlabd commented:
Thanks for the suggestions @mbstadler! I can limit the cutoff line by using `geom_segment` instead of `geom_hline`. I also changed the origin of the plot to match the data range. I've been struggling to limit the x-axis. Setting `xlim` also cuts the text we add with `ggrepel`.  Do you know if there's a good way around this?

Here is what I have done:
```
gg <- ggplot(data = df, 
       mapping = aes(x = regStep, y = selectionProbability)) +
  geom_line(mapping = aes(group = predictor, color = selected),
            linewidth = linewidth, 
            alpha = alpha) +
  scale_color_manual(values = c("TRUE" = selColor, 
                                "FALSE" = notSelColor)) +
  geom_segment(aes(x = min(regStep), xend = max(regStep) + 1, y = yintercept, yend = yintercept, linetype = cutoff),
               color = selProbCutoffColor, 
               linewidth = linewidth) +
  labs(x = "Regularization Step",
       y = "Selection Probability") +
  guides(colour = guide_legend(override.aes = list(alpha = 1))) +
  scale_y_continuous(expand = c(0, 0), limits = ylim) + 
  scale_x_continuous(expand = c(0, 0)) + 
  theme_classic()  

gg + ggrepel::geom_text_repel(data = subset(df, selected & regStep == max(regStep)), 
                              aes(label = predictor), nudge_x = 6, na.rm = TRUE, 
                              size = 3, direction = "y", color = "cadetblue", 
                              hjust = 0, segment.linetype = "dotted", 
                              segment.size = 0.7, segment.curvature = -0.1, 
                              segment.angle = 20, box.padding = 0.4, 
                              segment.alpha = 0.5, vjust = 0) + coord_cartesian(clip = 'off')
```
I will add this implementation following your suggestion, adding `ggrepel` to `Suggests`.
![stabSel](https://github.com/user-attachments/assets/aa0af961-5dcf-4bdd-8ad5-0c2fb38faf41)

machlabd committed:
added `.assertPackagesAvailable()` and
modified `plotStabilityPaths()` to include option to label predictors. Removed restriction that `ylim` needs to be in [0,1] to allow for more plotting flexibility.

machlabd committed:
fix ggplot2-related check notes by adding `.data$` to stability selection plot functions.

machlabd committed:
remove `graphics` dependency from `DESCRIPTION`

machlabd committed:
add `BiocManager` to `Suggests`

machlabd committed:
removed `subset` which was causing issues with `.data` use.

machlabd committed:
added `labelNudgeX` and `labelSize` for more flexibility when plotting labels in `plotStabilityPaths()`

machlabd committed:
test all plot functions work in the main vignette

machlabd commented on vignettes/monaLisa.Rmd:
We use the `plot`function here. I think it's ok to leave it, but in case you would also like to use `ggplot` here I wanted to point it out :) 
https://github.com/fmicompbio/monaLisa/blob/f4b3a71076ad920cee99995b40b5af08798f8561/vignettes/monaLisa.Rmd#L754

machlabd committed:
set no limit on `max.overlaps`for `plotStabilityPaths()`

machlabd committed:
add examples of `plotStabilityPaths()` with labels in the vignette

machlabd committed:
attempt to limit x-axis labels and ticks in `plotStabilityPaths()`

csoneson commented:
> I've been struggling to limit the x-axis. Setting xlim also cuts the text we add with ggrepel. Do you know if there's a good way around this?

I wonder if one way would be to use
```
coord_cartesian(xlim = ..., clip = "off")
```
I think this should allow the labels to go outside of the plotting area. It may require moving the legend somewhere else though, as they may collide.

machlabd commented:
# #

> > I've been struggling to limit the x-axis. Setting xlim also cuts the text we add with ggrepel. Do you know if there's a good way around this?
> 
> I wonder if one way would be to use
> 
> ```
> coord_cartesian(xlim = ..., clip = "off")
> ```
> 
> I think this should allow the labels to go outside of the plotting area. It may require moving the legend somewhere else though, as they may collide.

mm I tried this and it wasn't working for me. Following the current version of the plot function, and running the example from the help function, this is still clipping the text:
```
## create data set
Y <- rnorm(n = 500, mean = 2, sd = 1)
X <- matrix(data = NA, nrow = length(Y), ncol = 50)
for (i in seq_len(ncol(X))) {
  X[ ,i] <- runif(n = 500, min = 0, max = 3)
}
s_cols <- sample(x = seq_len(ncol(X)), size = 10,
  replace = FALSE)
for (i in seq_along(s_cols)) {
  X[ ,s_cols[i]] <- X[ ,s_cols[i]] + Y
}

## reproducible randLassoStabSel() with 1 core
set.seed(123)
ss <- randLassoStabSel(x = X, y = Y)

## plot
plotStabilityPaths(ss, labelPaths = TRUE) + coord_cartesian(xlim = c(1,40), clip = "off")
```

mbstadler commented:
How about suppressing the x-axis line with `theme(axis.line.x = element_blank())` and then manually drawing it with `geom_segment`, like:
```
plotStabilityPaths(ss, labelPaths = TRUE) +
    geom_segment(data = data.frame(x = 1, xend = 41, y = 0),
                 mapping = aes(x = x, xend = xend, y = y, yend = y)) +
    theme(axis.line.x = element_blank())
```
(of course the x-range would have to be calculated just as for the `geom_segment` that draws the cutoff line.

machlabd committed:
fix the x-axis for `plotStabilityPaths()` following Michael's suggestion

machlabd commented:
> How about suppressing the x-axis line with `theme(axis.line.x = element_blank())` and then manually drawing it with `geom_segment`, like:
> 
> ```
> plotStabilityPaths(ss, labelPaths = TRUE) +
>     geom_segment(data = data.frame(x = 1, xend = 41, y = 0),
>                  mapping = aes(x = x, xend = xend, y = y, yend = y)) +
>     theme(axis.line.x = element_blank())
> ```
> 
> (of course the x-range would have to be calculated just as for the `geom_segment` that draws the cutoff line.

That works, thanks! I have added the `geom_segment` implementation to both the x- and y-axis so that line widths are matching as well. Currently the labels are printed in a random order if they end on the same y-value. I'll mention that in the vignette as well.

machlabd committed:
mention stochasticity if labels have the same y-value in `plotStabilityPaths()`

machlabd committed:
started updating `NEWS.md` file

csoneson committed:
Make indentation and spacing consistent

csoneson committed:
Update plotStabilityPaths checks (no need to check explicitly for rownames(colData(se)), as they are the same as colnames(se))

csoneson committed:
Add tests for .assertPackagesAvailable

csoneson committed:
Expand test coverage for plotStabilityPaths

csoneson committed:
Expand test coverage for binning

csoneson committed:
Add tests for .assertColor, make indentation and spacing consistent with the rest of the file

csoneson committed:
Make .assertColor internal

csoneson committed:
Update badges

csoneson committed:
Update bin plot in README

csoneson committed:
Minor text change in vignette

csoneson committed:
Bump version

machlabd committed:
Captalize start of parameter descriptions in `plotStabilityPaths()`, matching the rest of the funcitons

machlabd committed:
more aesthetic compatibility fixes in `plotStabilityPaths()`, added `\code` on TRUE.

machlabd committed:
added `rlang` import to stability selection plot functions

machlabd committed:
checked capitalization if param description start for all plot functions

machlabd committed:
Fix capitalization in all plot functions.

machlabd committed:
added some missing `\code{}`.

machlabd commented on R/stability_selection.R:
```diff
@@ -90,23 +91,30 @@
-#' @param cutoff value between 0 and 1 (default = 0.8) which is the cutoff
+#' @param glmnet.args Named list with additional arguments (beyond \code{x}, 
+#'     \code{y} and \code{weakness}, which are determined automatically, and 
+#'     \code{q} that can be passed via \code{...}) that 
```
Should we also mention the `type` argument in `.glmnetRandomizedLasso()`?

csoneson replied:
Thanks - the reason I didn't mention `type` specifically was that they way I read the code, this would neither be automatically determined nor be passed via `...` (since it's not an argument to `stabs::stabsel()`). So I think it could be provided to `glmnet.args` (and thus is not an exception). Did I misunderstand?

machlabd commented on R/stability_selection.R:
```diff
@@ -90,23 +91,30 @@
+#'     will be passed to the internal \code{.glmnetRandomizedLasso} function. 
+#'     The available arguments to the latter are the same as the ones for 
+#'     \code{\link[stabs]{glmnet.lasso}}. A typical use case would be to define 
+#'     the \code{family} argument to \code{glmnet.lasso}. 
```
The arguments are passed on to the `glmnet::glmnet`function which is used in `.glmnetRandomizedLasso()`. So I would phrase it as: "The available arguments to the latter are the same as the ones for \code{\link[glmnet]{glmnet}}. "

csoneson replied:
Fixed

csoneson replied:
Although this now doesn't cover `q` and `type`, and the `pmax`, `dfmax` and `penalty.factor` are also automatically determined...

machlabd replied:
Thanks for the fixes! So `q` and `type` are arguments for `stabs::stabsel`. Given the set values for `cutoff` and `PFER`, `q` will be estimated by the function. So then the only arguments that get determined automatically should be `q` in `stabs::stabsel`, and the rest of the arguments in `glmnet::glmnet` like `penalty.factor`, `pmax` and `dfmax`, though they are set in a specific way for stability selection (related to `q` and `weakness`).

Sorry if I misunderstood.

On another note, currently we do not allow for the flexibilty of the user providing `q` instead of `PFER` for example, to run stability selection. If `q` is set by the user, the `stabs::stabsel` will complain that all three argumetns have been set and that it should be limited to two of them. Do you think we should change that or keep it as is for now?

machlabd replied:
Ah sorry `type` is not an argument for `stabsel`! Sorry for the confusion there. Please ignore that! You are right, `q` and `type` are set automatically and by default respectively. And it's nice to distinguish the `glmnet` args that are set automatically which we do.

csoneson replied:
I just discussed quickly with Michael and we changed the formulation in the documentation slightly (see https://github.com/fmicompbio/monaLisa/pull/86/commits/05df5079008f08cc6acd1f5db808a9cefd1cd8d1). We felt that this might be the most correct, perhaps least confusing, way of formulating the requirements - what do you think?

machlabd replied:
I think this is very clear now. Thank you both!

machlabd commented on R/stability_selection.R:
```diff
@@ -218,36 +226,48 @@
+        if (any(i <- names(glmnet.args) %in% c("x", "y", "q", "weakness"))) {
+            warning(paste0("Ignoring the following elements of 'glmnet.args' (as ", 
+                           "they are set automatically): ", 
+                           paste(names(glmnet.args)[i], collapse = ",")))    
```
maybe here the check should be that they are valid arguments in `glmnet::glmnet`

csoneson replied:
I added this check

csoneson committed:
Update x-axis label in README density plot

csoneson committed:
Fix to randLassoStabSel docs

csoneson committed:
Check that glmnet.args are valid arguments to glmnet or glmnet.lasso

mbstadler committed:
minor edits to NEWS

mbstadler committed:
Merge branch 'ggplot2switch' of github.com:fmicompbio/monaLisa into ggplot2switch

mbstadler committed:
add noise and both directions to example data

mbstadler committed:
add a bit of whitespace between label and bar

csoneson committed:
Update docs for randLassoStabSel

mbstadler committed:
use geom_col instead of geom_bar(..., stat)

mbstadler committed:
update docs

mbstadler committed:
Merge branch 'ggplot2switch' of github.com:fmicompbio/monaLisa into ggplot2switch

mbstadler committed:
use .data to avoid note about missing variables
